### ISSUE TITLE
Add option to add more options to docker storage

### ISSUE THREAD

**Issue by cgwalters** [projectatomic/docker-storage-setup#6]
If say I want to pass e.g. `--storage-opt dm.basesize=50G` to docker, currently the easiest way to do that is to change `/etc/sysconfig/docker`, where one would expect at least to also be able to change either `/etc/sysconfig/docker-storage-setup` where other options are specified.

Alternatively, we could figure out how to sanely merge human and automatic changes in `/etc/sysconfig/docker-storage`, but that's nontrivial, and I prefer not mixing human-edited and machine-edited files.

**Comment by sdodson** [comment]
would you expect this to also address adding version specific flags automatically? For instance docker-1.9.1 added `dm.use_deferred_deletion=true` and it'd be nice to get that during upgrades by merging machine generated and user modified.

**Comment by rhvgoyal** [comment]
Now we have a config option "EXTRA_DOCKER_STORAGE_OPTIONS", where one can specify the config options to pass to docker.

**Comment by sdodson** [comment]
So this issue can be closed then, right? 

Probably a different issue, but If I upgrade from docker-1.8.2 to docker-1.9.1, assuming d-s-s runs when docker starts, will I get the new options defined?

**Comment by rhvgoyal** [comment]
I think this particular issue can be closed. What you are asking for is a different issue.

I think right fix for your issue is to enable deferred deletion and deferred removal by default in docker that way over an upgrade customer automatically gets it.

Now docker has done changes in their systemd unit file to run docker daemon in host namespace by default. That means lot of leaking of mounts across namespaces and that also means lot more need to enable deferred deletion and deferred removal by default. So may be docker will be open to it now. If not, we should think of carrying a patch for docker for fedora.

**Comment by rhvgoyal** [comment]
cc @shishir-a412ed Would you like to propose enabling deferred removal and deferred deletion by default upstream?

**Comment by rhatdan** [comment]
Yes @shishir-a412ed propose that and explain the leaking of open file descriptors caused by running docker in the host namespace.

**Comment by shishir-a412ed** [comment]
@rhvgoyal @rhatdan ack. Will open a PR upstream.

**Comment by shishir-a412ed** [comment]
@rhvgoyal @rhatdan 
https://github.com/docker/docker/pull/23590

Shishir
